waiter/container_delete.go

```diff
@@ -0,0 +1,73 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+// ContainerDeleteExecutor represents requirements to async container delete operation.
+type ContainerDeleteExecutor interface {
+	ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error
+	ContainerGet(ctx context.Context, id cid.ID, prm client.PrmContainerGet) (container.Container, error)
+}
+
+// ContainerDeleteWaiter implements sync logic to container delete operation.
+type ContainerDeleteWaiter struct {
+	executor     ContainerDeleteExecutor
+	pollInterval time.Duration
+}
+
+// NewContainerDeleteWaiter is a constructor for ContainerDeleteWaiter.
+func NewContainerDeleteWaiter(executor ContainerDeleteExecutor) ContainerDeleteWaiter {
+	return ContainerDeleteWaiter{executor: executor, pollInterval: defaultPollInterval}
+}
+
+// SetPollInterval allows rewrite default poll interval.
+func (w *ContainerDeleteWaiter) SetPollInterval(interval time.Duration) {
+	w.pollInterval = interval
+}
+
+// ContainerDelete sends request to remove the NeoFS container.
+//
+// See details in [client.Client.ContainerDelete].
```

Comment:
Eventually we call the client. I wanted to left link to the documentation. May be I should change description to `See documentation for function`

Reply:
> Eventually we call the client.

in our code. but a user can pass any `ContainerDeleteExecutor` implementation so i doubt we can document it like that

Reply:
We can reference the interface here then (whatever is passed into the waiter is used). And that can reference the client (because that's the semantics we expect in the waiter).

Reply:
Updated docs